script.js

```diff
@@ -1,4 +1,251 @@
+  if (score > 21) {
+    score = score - 10;
+  }
+  return score;
+}
+
+function displayPlayerHandAndScore(player, handArray, score, gameMode) {
+  message = "";
+
+  if (gameMode === "startGame") {
+    message = `${player}'s Hand: ${listCards(player, handArray, gameMode)}`;
+  } else {
+    message = `${player}'s Hand: ${listCards(
+      player,
+      handArray,
+      gameMode
+    )} Score: ${score}`;
+  }
+
+  if (isBust(handArray)) {
+    message = message + " (BUST!)";
+  }
+  message = message + "<br><br>";
+  return message;
+}
+
+function listCards(player, handArray, gameMode) {
+  var cardList = "";
+  if (player === "Dealer" && gameMode === "startGame") {
+    cardList =
+      cardList +
+      `- ${handArray[0].name} of ${handArray[0].suit}<br> - (Hidden Card)<br>`;
+  } else {
+    for (i = 0; i < handArray.length; i += 1) {
+      cardList =
+        cardList + `- ${handArray[i].name} of ${handArray[i].suit}<br>`;
+    }
+  }
+  return `<br> ${cardList}`;
+}
+
+function isBust(handArray) {
+  score = calculateScore(handArray);
+  return score > 21;
+}
+
+function generateResult(playerHandArray, dealerHandArray) {
+  playerScore = calculateScore(playerHandArray);
+  dealerScore = calculateScore(dealerHandArray);
+  gameResult = "";
+  if (isBust(dealerHandArray) && isBust(playerHandArray)) {
+    gameResult = "Tie!";
+  } else if (isBust(dealerHandArray)) {
+    gameResult = "Player wins!";
+  } else if (isBust(playerHandArray)) {
+    gameResult = "Dealer wins!";
+  } else if (playerScore == dealerScore) {
+    gameResult = "Tie!";
+  } else if (dealerScore > playerScore) {
+    gameResult = "Dealer  wins!";
+  } else {
+    gameResult = "Player wins!";
+  }
+  return gameResult;
+}
+
+//--- MAIN FUNCTION ---
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if (gameMode === "startGame") {
+    //game set up: create deck, shuffle deck and deal two cards to player and dealer.
+    dealerHand = [];
+    playerHand = [];
+    cardDeck = makeDeck();
+    cardDeck = shuffleCards(cardDeck);
+
+    dealerHand = dealCards(dealerHand, cardDeck, 2);
+    dealCards(playerHand, cardDeck, 2);
+    dealerScore = calculateScore(dealerHand);
+    playerScore = calculateScore(playerHand);
+
+    myOutputValue =
+      `You received two cards. <br><br>` +
+      displayPlayerHandAndScore("Player", playerHand, playerScore, gameMode) +
+      displayPlayerHandAndScore("Dealer", dealerHand, dealerScore, gameMode);
+
+    // if either player gets blackjack, game ends
+    if (isBlackjack(playerHand) || isBlackjack(dealerHand)) {
+      if (isBlackjack(playerHand) && isBlackjack(dealerHand)) {
+        myOutputValue =
+          myOutputValue + "WOOWW YOU BOTH GOT BLACKJACK! It's a tie!";
+      } else if (isBlackjack(playerHand)) {
+        myOutputValue = myOutputValue + "BLACKJACK! You win!";
+      } else if (isBlackjack(dealerHand)) {
```

Comment:
This could be an `else` statement. You generally want to have an else statement. Reason being, with else if statements, there is a possibility that none of your conditions will be met. But having an else statement guarantees that at least the else statement will run